lib/puppet/provider/keycloak_client/kcadm.rb

```diff
@@ -243,6 +277,23 @@ def create
         raise Puppet::Error, "kcadm update clients/#{resource[:id]}/optional-client-scopes/#{scope_id}: #{e.message}"
       end
     end
+    if resource[:roles]
+      roles = @property_hash[:roles] ||= get_client_roles(resource[:realm], resource[:id])
```

Comment:
I think my previous recommendation was wrong.  `@property_hash` won't be set during `create` because `create` is called when something does not exist.  So your previous approach was correct, only do this:

```
role = get_client_roles(resource[:realm], resource[:id])
```

Also any reason not to move this logic under same `if resource[:roles]` ? That would make the code a little less duplicated.